src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +65,61 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
             return this.configurationFactory.CreateAnyCallConfiguration(manager, rule);
         }
 
+        public IVoidArgumentValidationConfiguration CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
+            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+
+            var setterExpression = BuildSetterFromGetter(propertySpecification);
+            var fake = this.expressionParser.GetFakeManagerCallIsMadeOn(setterExpression);
+            var rule = this.ruleFactory.Invoke(setterExpression);
+
+            fake.AddRuleFirst(rule);
+
+            return this.configurationFactory.CreateConfiguration(fake, rule);
+        }
+
+        private string GetExpressionDescription<TValue>(Expression<Func<TValue>> expression)
+        {
+            var matcher = new ExpressionCallMatcher(
+                expression,
+                ServiceLocator.Current.Resolve<ExpressionArgumentConstraintFactory>(),
+                ServiceLocator.Current.Resolve<MethodInfoManager>(),
+                this.callExpressionParser);
+
+            var expressionDescription = matcher.DescriptionOfMatchingCall;
+            return expressionDescription;
+        }
+
+        private LambdaExpression BuildSetterFromGetter<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            MemberExpression memberExpression = propertySpecification.Body as MemberExpression;
+            if (memberExpression == null)
+            {
+                var expressionDescription = GetExpressionDescription(propertySpecification);
+                throw new ArgumentException("Expression '" + expressionDescription + "' must refer to a property getter, but doesn't.");
+            }
+
+            PropertyInfo propertyInfo = memberExpression.Member as PropertyInfo;
+            if (propertyInfo == null)
+            {
+                throw new ArgumentException("The expression refers to '" + memberExpression.Member.Name + "', which is a field, not a property getter.");
+            }
+
+            var propertySetterInfo = propertyInfo.GetSetMethod(nonPublic: true);
+            if (propertySetterInfo == null)
+            {
+                throw new ArgumentException("The property '" + memberExpression.Member.Name + "' does not have a setter.");
+            }
+
+            Expression<Func<TValue>> value = () => A<TValue>.Ignored;
+
+            IEnumerable<Expression> indices = propertySpecification.Parameters.Cast<Expression>();

```

Comment:
`Parameters` is the list of parameters of the lambda, not the property's indices. In this case it's always empty, because an `Func<TValue>` doesn't have any parameter.